app/api/public/users/route.ts

```diff
@@ -0,0 +1,199 @@
+import { NextRequest, NextResponse } from "next/server";
+import prisma from "../../../../lib/prisma";
+import { TIERS } from "../../../../lib/constants";
+import { fetchFarcasterProfiles, UserProfile } from "../../../../lib/airstack";
+import { formatAirstackUserData } from "../../../../lib/airstack/utils";
+import { WaitlistTier } from "@prisma/client";
+
+export const POST = async (req: NextRequest) => {
+  try {
+    // Get the API key from the headers
+    const apiKey = req.headers.get("x-api-key");
+
+    if (!apiKey) {
+      return NextResponse.json(
+        {
+          message: "Unauthorized",
+        },
+        {
+          status: 401,
+        }
+      );
+    }
+
+    // Check if the API key exists in the database
+    const key = await prisma.apiKey.findUnique({
+      where: {
+        key: apiKey,
+      },
+    });
+
+    // If the API key doesn't exist, return a 404
+    if (!key || key.mode === "r") {
+      return NextResponse.json(
+        {
+          message: "Unauthorized",
+        },
+        {
+          status: 401,
+        }
+      );
+    }
+
+    // Create an API request log in the database
+    await prisma.apiRequest.create({
+      data: {
+        api_key_id: key.id,
+        path: req.url.replace(`${process.env.BASE_URL}`, ""),
+        method: "POST",
+        createdAt: new Date(),
+      },
+    });
+
+    // get the x-waitlist-id from the headers that surely exists because of the middleware
+    const waitlistId = key.waitlist_id;
+
+    // Find the waitlist associated with the id
+    const waitlist = await prisma.waitlist.findUnique({
+      where: {
+        id: waitlistId,
+      },
+      include: {
```

Comment:
Let's remove this _count check, as we (for now) allow only QUEEN waitlist to use the API, so no limit on users to add

Reply:
Should we also disable the checkbox if the waitlist is not a QUEEN one?
Also: should we add a line in the readme and in the pricing page to state this new feature?